pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -642,31 +668,35 @@ var RevVMImages = func() map[string]echo.VMDistro {
 	return r
 }()
 
-func templateParams(cfg echo.Config, settings *resource.Settings) (map[string]interface{}, error) {
-	if settings == nil {
-		var err error
-		settings, err = resource.SettingsFromCommandLine("template")
-		if err != nil {
-			return nil, err
-		}
+// getVMOverrideForIstiodDNS returns the DNS alias to use for istiod on VMs. VMs always access
+// istiod via the east-west gateway, even though they are installed on the same cluster as istiod.
+func getVMOverrideForIstiodDNS(ctx resource.Context, cfg echo.Config) (istioHosts []string, istioIP string) {
+	if ctx == nil {
+		return
 	}
 
-	supportStartupProbe := cfg.Cluster.MinKubeVersion(0)
+	ist, err := istio.Get(ctx)
+	if err != nil {
+		log.Warnf("VM config failed to get Istio component for %s: %v", cfg.Cluster.Name(), err)
+		return
+	}
 
-	vmImage := VMImages[cfg.VMDistro]
-	_, knownImage := RevVMImages[cfg.VMDistro]
-	if vmImage == "" {
-		if knownImage {
-			vmImage = cfg.VMDistro
-		} else {
-			vmImage = VMImages[echo.DefaultVMDistro]
-		}
-		log.Debugf("no image for distro %s, defaulting to %s", cfg.VMDistro, echo.DefaultVMDistro)
+	systemNS := ist.Settings().SystemNamespace
+	istioHosts = []string{
+		fmt.Sprintf("istiod.%s.svc.cluster.local", systemNS),
+		fmt.Sprintf("istiod.%s", systemNS),
+		fmt.Sprintf("istiod.%s.svc", systemNS),
```

Comment:
nit: kind of ties in with my other comment, but I think we _only_ want istiod.%s.svc, or whatever istioctl generates.

An example bug we would miss now is if the agent was making a call to istiod.%s; this would fail on a  real VM but pass here.